src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java

```diff
@@ -555,6 +625,23 @@ private static void handleKeyValueConfig(TestInputConfiguration.Builder testInpu
         }
     }
 
+    public static Map<String, String> getDefaultProperties(String filePath, String moduleName)
+            throws CheckstyleException {
+
+        final String fullyQualifiedClassName = getFullyQualifiedClassName(filePath, moduleName);
+        final ModuleDetails moduleDetails = MODULE_DETAILS_MAP.get(fullyQualifiedClassName);
+
+        List<ModulePropertyDetails> properties = moduleDetails.getProperties();
+        if (properties == null) {
+            properties = Collections.emptyList();
+        }
+
+        final Map<String, String> result = new HashMap<>();
+        properties.forEach(prop -> result.put(prop.getName(), prop.getDefaultValue()));
+
+        return result;
+    }
```

Comment:
```suggestion
            return Optional.ofNullable(moduleDetails.getProperties())  
                        .orElseGet(Collections::emptyList)  
                        .stream()  
                        .collect(Collectors.toMap(  
                            ModulePropertyDetails::getName,  
                            ModulePropertyDetails::getDefaultValue,  
                            (oldValue, newValue) -> newValue  
                        )); 
```
try to reduce overhead and make simpler functional approach leveraging streaming api.

Reply:
There are checks which have properties whose default value is null, and therefore  a NPE is thrown if we use `Collectors.toMap()` .